### ISSUE TITLE
False positives for InetAddresses.isInetAddress

### ISSUE THREAD

**Issue by devikasondhi** [google/guava#3116]
InetAddresses.isInetAddress method returns true for invalid IPv6 address containing two or more consecutive 0s in the last hextet.
Failing test case:
InetAddresses.isInetAddress("804c:404c::0:cc:c0:00c1c") //returns true

Expected to return false, as also verified with an online service here- https://www.helpsystems.com/intermapper/ipv6-test-address-validation

The issue seems to be due to the compression over zeroes involved.

**Comment by pmarks-net** [comment]
You're suggesting that `parseHextet(String ipPart)` should fail when `ipPart.length() > 4`.

I don't have a strong preference either way, but by what definition is this invalid?  Guava isn't the only parser that works like this:

    $ ping6 ::00000001
    PING ::00000001(::1) 56 data bytes
    64 bytes from ::1: icmp_seq=1 ttl=64 time=0.049 ms

**Comment by swankjesse** [comment]
The [RFC](https://tools.ietf.org/html/rfc4291#section-2.2) suggests both Guava and `ping6` have it wrong here: 

>    1. The preferred form is x:x:x:x:x:x:x:x, where the 'x's are one to
>       four hexadecimal digits of the eight 16-bit pieces of the address.

**Comment by pmarks-net** [comment]
Looking at the history of the document:

* [RFC1884](https://tools.ietf.org/html/rfc1884#section-2.2) (1995): "where the 'x's are the hexadecimal values of the eight 16-bit pieces of the address."
* [RFC2373](https://tools.ietf.org/html/rfc2373#section-2.2) (1998): same
* [RFC3513](https://tools.ietf.org/html/rfc3513#section-2.2) (2003): same
* [RFC4291](https://tools.ietf.org/html/rfc4291#section-2.2) (2006): "where the 'x's are one to four hexadecimal digits of the eight 16-bit pieces of the address."

So the current behavior was standard for more than a decade, and is consistent with OpenJDK:

    System.err.println("ip=" + InetAddress.getByName("::00000001"));
    ...
    ip=/0:0:0:0:0:0:0:1

In other words, I don't see a compelling reason to change it.

**Comment by sandhu5** [comment]
[Test.java.zip](https://github.com/google/guava/files/2076902/Test.java.zip)
